gpt4all-chat/chatllm.cpp

```diff
@@ -350,6 +350,7 @@ bool ChatLLM::loadModel(const ModelInfo &modelInfo)
                 }
 
                 m_llModelInfo.model->setProgressCallback([this](float progress) -> bool {
+                    progress = std::max(progress, std::numeric_limits<float>::min()); // keep progress above zero
```

Comment:
Can model loading fail before the first possible time this callback is called? If so, this seems like a hack as the proper progress is then 0, right?

Reply:
The sole purpose of this line is to prevent the initial call of the progress callback, which seems to always have progress=0, from resetting the state of the progress bar.